deepchem/models/tensorgraph/layers.py

```diff
@@ -156,6 +171,50 @@ def add_summary_to_tg(self):
     elif self.summary_op == 'histogram':
       tf.summary.histogram(self.name, self.tb_input, self.collections)
 
+  def copy(self, replacements={}, variables_graph=None):
+    """Duplicate this Layer and all its inputs.
+
+    This creates and returns a clone of this layer.  It also recursively calls
+    copy() on all of this layer's inputs to clone the entire hierarchy of layers.
+    In the process, you can optionally tell it to replace particular layers with
+    specific existing ones.  For example, you can clone a stack of layers, while
+    connecting the topmost ones to different inputs.
```

Comment:
Would you mind adding in a code example here in the docstring? This function's usage is complicated enough a little bit of code would really help here

Reply:
This is a basic question, but which `tf.Graph` do the copied layers exist in? Is it the same graph as the original model or in a new graph?

Reply:
It's copying the layers, not the tensors, so they aren't specific to any graph.  If you then add the copied layers to a TensorGraph, its `build()` method will create tensors in that graph.

Reply:
Ah, that makes sense.